chia/full_node/coin_store.py

```diff
@@ -411,6 +411,95 @@ async def get_coin_states_by_ids(
 
         return coins
 
+    async def batch_coin_states_by_puzzle_hashes(
+        self,
+        puzzle_hashes: List[bytes32],
+        *,
+        min_height: uint32 = uint32(0),
+        include_spent: bool = True,
+        include_unspent: bool = True,
+        include_hinted: bool = True,
+        max_items: int = 50000,
+    ) -> Tuple[List[CoinState], Optional[uint32]]:
+        """
+        Returns the coin states, as well as the next block height (or `None` if finished).
+        Note that the maximum number of puzzle hashes is currently set to 15000.
+        """
+
+        assert len(puzzle_hashes) <= 15000
```

Comment:
Should this be hard-coded like this? Perhaps a config item - or maybe even a part of the protocol version in some way? hard-coded numbers just tend to be problems eventually

Reply:
Currently this code won't work if the limit is above half of the SQLite variable limit (which sometime in Python 3.8 the SQLite version was raised such that the limit is around 32,700, rather than 999). I haven't found a performant enough solution that doesn't involve using variables like this.

That's why this number is fixed to this value, since raising it could cause issues in the query (I believe it would just ignore the rest or something like that). I could make it a constant or something though, maybe.

I am implementing the protocol in such a way that this limit is mostly irrelevant and can change at any time without breaking the protocol (it will return the ignored puzzle hashes so you can do it in multiple passes on the wallet side if handled properly).

Reply:
Perhaps some comment in the code that this limit is related to the sqlite parameter limit